src/bin/bobp.rs

```diff
@@ -1,150 +1,494 @@
+        });
+        let res = if measure_time {
+            let start = Instant::now();
+            let res = client.get(request).await;
+            stat.save_single_thread_get_time(&start.elapsed());
+            res
+        } else {
+            client.get(request).await
+        };
         if res.is_err() {
             stat.get_error.fetch_add(1, Ordering::SeqCst);
         }
         stat.get_total.fetch_add(1, Ordering::SeqCst);
     }
 }
 
-async fn put_worker(net_conf: NetConfig, task_conf: TaskConfig, stat: Arc<Stat>) {
-    let mut client = build_client(net_conf.clone()).await;
-
-    let options: Option<PutOptions> = if task_conf.direct {
-        Some(PutOptions {
-            remote_nodes: vec![],
-            force_node: true,
-            overwrite: false,
-        })
-    } else {
-        None
-    };
+async fn put_worker(net_conf: NetConfig, task_conf: TaskConfig, stat: Arc<Statistics>) {
+    let mut client = net_conf.build_client().await;
 
+    let options: Option<PutOptions> = task_conf.find_put_options();
+    let measure_time = task_conf.is_time_measurement_thread();
     let upper_idx = task_conf.low_idx + task_conf.count;
     for i in task_conf.low_idx..upper_idx {
-        let meta = BlobMeta {
-            timestamp: SystemTime::now()
-                .duration_since(UNIX_EPOCH)
-                .expect("msg: &str")
-                .as_secs() as i64,
-        };
-        let blob = Blob {
-            data: vec![0; task_conf.payload_size as usize],
-            meta: Some(meta),
-        };
+        let blob = create_blob(&task_conf);
         let key = BlobKey { key: i };
         let req = Request::new(PutRequest {
             key: Some(key),
             data: Some(blob),
             options: options.clone(),
         });
-        let res = client.put(req).await;
+        let res = if measure_time {
+            let start = Instant::now();
+            let res = client.put(req).await;
+            stat.save_single_thread_put_time(&start.elapsed());
+            res
+        } else {
+            client.put(req).await
+        };
         if res.is_err() {
             stat.put_error.fetch_add(1, Ordering::SeqCst);
         }
         stat.put_total.fetch_add(1, Ordering::SeqCst);
     }
 }
 
-#[tokio::main]
-async fn main() {
-    let matches = App::new("Bob benchmark tool")
+async fn test_worker(net_conf: NetConfig, task_conf: TaskConfig, stat: Arc<Statistics>) {
+    put_worker(net_conf.clone(), task_conf.clone(), stat.clone()).await;
+    get_worker(net_conf.clone(), task_conf.clone(), stat.clone()).await;
+}
+
+async fn ping_pong_worker(net_conf: NetConfig, task_conf: TaskConfig, stat: Arc<Statistics>) {
+    let mut client = net_conf.build_client().await;
+
+    let get_options = task_conf.find_get_options();
+    let put_options = task_conf.find_put_options();
+    let measure_time = task_conf.is_time_measurement_thread();
+    let upper_idx = task_conf.low_idx + task_conf.count;
```

Comment:
what if low_idx set to 0 in config?

Reply:
We'll iterate from 0 to count

Reply:
I made a mistake, opposite - what if low_idx > 0 in config. There will be no time measurement threads?

Reply:
Yep, currently there will be no time measuring... I'll fix this